### ISSUE TITLE
Decorator functions using ParamSpec cause errors when used on classmethods

### ISSUE THREAD

**Issue by bijij** [microsoft/pyright#1939]
**Describe the bug**
classmethods decorated by a function using `ParamSpec` cause an error when attempting to access said classmethod.

**To Reproduce**
Given the example code shown below: `test.py`

Pyright will exclaim the following as one would expect; 

`test.py:25:21 - info: Type of "self.randint" is "(min: int = ..., max: int = ...) -> int"`
`test.py:28:13 - info: Type of "Foo.randint" is "(self: Foo, min: int = ..., max: int = ...) -> int"`

However by appending the `classmethod` decorator to `Foo.randint` pyright will return the following;

`test.py:26:21 - info: Type of "self.randint" is "(min: int = ..., max: int = ...) -> int"`
`test.py:27:21 - error: Could not bind method "" because "Foo" is not assignable to parameter "cls"
    "Foo" is incompatible with "Type[Foo]" (reportGeneralTypeIssues)`
`test.py:29:17 - error: Could not bind method "" because "Type[Foo]" is not assignable to parameter "cls"       
    "Foo" is incompatible with "Type[Foo]" (reportGeneralTypeIssues)`
`test.py:29:13 - info: Type of "Foo.randint" is "(cls: Type[Foo], min: int = ..., max: int = ...) -> int"`

note: `self` was renamed to `cls` but this does not matter.

**Expected behavior**
No errors would occur when accessing the classmethod `Foo.randint`

**Screenshots or Code**
Conscider the following python 3.9 code: `test.py`

```python
from __future__ import annotations

import random
from typing import TYPE_CHECKING, Callable

if TYPE_CHECKING:
    from typing_extensions import ParamSpec

    P = ParamSpec("P")


def rounder(func: Callable[P, float]) -> Callable[P, int]:
    def wrapper(*args: P.args, **kwargs: P.kwargs) -> int:
        return round(func(*args, **kwargs))

    return wrapper


class Foo:
    @rounder
    def randint(self, min: int = 0, max: int = 10) -> float:
        return random.random() * (max - min) + min

    def test(self) -> int:
        reveal_type(self.randint)
        return self.randint()
        
reveal_type(Foo.randint)
```

Which pyright sucesfully type-checks:

![image](https://user-images.githubusercontent.com/8677174/120634220-5ab38480-c4ae-11eb-9ba2-967e7b11afd0.png)

When the changes outlined in the steps above are applied the following occurs:

![image](https://user-images.githubusercontent.com/8677174/120634267-656e1980-c4ae-11eb-943f-87af7f10334d.png)

**VS Code extension or command-line**
Tested using both Pylance 2021.6.0 and Pyright CLI 1.1.145

**Additional context**
Tested on both python 3.9 using deferred evaluation of annotations and python 3.10 directly.

Interestingly, swapping the order of the decorators causes the same issue, I would have expected however this would cause a different error given it would error at runtime (`classmethod` objects are not callable).

**Comment by erictraut** [comment]
Thanks for the bug report. The ParamSpec code was not properly tracking the fact that a method was marked as a classmethod or a staticmethod. I've updated the code to handle this correctly. The fix will be in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.147, which I just published. It will also be included in the next release of pylance.
